exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -125,7 +154,7 @@ protected void cancel() {
   // (Not delegated.)
   @Override
   public boolean next() throws SQLException {
-    throwIfClosed();
+    throwIfTimedOutOrClosed();
```

Comment:
Tested this out with the latest commit and I'm able to successfully interrupt the next() as well. Occasionally, the exceptions get missed, so we do a check down to the DrillResultSet and not the DrillCursor as I'm trying to restrict the timeout check only to the JDBC layers and not too close to the DrillClient. Testcases verify by executing and then sleeping till the duration of the timeout before invoking the next() method.